### ISSUE TITLE
Since version 4.2.0 observe non confirmable not working properly

### ISSUE THREAD

**Issue by vincent-iQontrol** [shamblett/coap#107]
I use a non confirmable observer from the example. Version 4.0.0 and 4.1.0 is working correctly, all events are coming in. In 4.2.0 and above a lot of events are missing after ~ 13 events.

final reqObsNon = CoapRequest(CoapCode.get, confirmable: false)..addUriPath('Json');

**Comment by JosefWN** [comment]
Confirmed using this snippet (works in 4.1.0 but not in 4.2.0):
```dart
import 'dart:async';
import 'package:coap/coap.dart';
import 'config/coap_config.dart';

FutureOr main() async {
  final conf = CoapConfig();
  final uri = Uri(
    scheme: 'coap',
    host: 'californium.eclipseprojects.io',
    port: conf.defaultPort,
  );
  final client = CoapClient(uri, conf);

  try {
    final reqObsNon = CoapRequest(CoapCode.get, confirmable: false);
    reqObsNon.addUriPath('obs-non');

    print('Observing /obs-non on ${uri.host}');
    final obsNon = await client.observe(reqObsNon);
    obsNon.stream.listen((CoapRespondEvent e) {
      print('/obs-non response: ${e.resp.payloadString}');
    });

    await Future.delayed(Duration(minutes: 15));
  } catch (e) {
    print('CoAP encountered an exception: $e');
  }

  client.close();
}
```

I will try to dig into it soon.

**Comment by vincent-iQontrol** [comment]
Thanks for investigating. I will use an older version for the time being.

**Comment by JosefWN** [comment]
Made a couple of more test, this bug seems to have been introduced with my stricter linting of the repo...

**Comment by shamblett** [comment]
Yep ok Ill look at the PRs over the weekend ta.

**Comment by shamblett** [comment]
With this and #112 now merged are we ready for another release? There's quite a bitt gone since 4.2.1

**Comment by JosefWN** [comment]
We could merge PR #111 too perhaps? Then I can rebase my other PR on master to reduce its size.

**Comment by JKRhb** [comment]
> With this and #112 now merged are we ready for another release? There's quite a bitt gone since 4.2.1

I also have one or two more upcoming PRs which I will open soon, then we could go for a new release :)

**Comment by shamblett** [comment]
OK PR #111 merged, I'll hold off on the release.

**Comment by JKRhb** [comment]
> 
> I also have one or two more upcoming PRs which I will open soon, then we could go for a new release :)

Okay, I think I've now opened all PRs I had in flight :) Due to the fact that some of our PRs contain breaking changes, should we go for a 5.0.0 release?

**Comment by shamblett** [comment]
Believe this has been addressed. Package re released at version 5.0.0.

**Comment by shamblett** [comment]
Package now re published at version 6.0.0, closing this issue.
